<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>EightPuzzleSolver</class>
 <widget class="QMainWindow" name="EightPuzzleSolver">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>404</width>
    <height>373</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>EightPuzzleSolver</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QGridLayout" name="gridLayout_11">
    <item row="0" column="1">
     <widget class="QFrame" name="frame">
      <property name="frameShape">
       <enum>QFrame::StyledPanel</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Raised</enum>
      </property>
      <widget class="QWidget" name="layoutWidget">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>10</y>
         <width>152</width>
         <height>284</height>
        </rect>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_7">
        <item>
         <widget class="QGroupBox" name="groupBox_5">
          <property name="title">
           <string>The start</string>
          </property>
          <layout class="QGridLayout" name="gridLayout_12">
           <item row="0" column="0">
            <layout class="QVBoxLayout" name="verticalLayout_5">
             <item>
              <layout class="QGridLayout" name="gridLayout_15">
               <item row="2" column="1">
                <widget class="QLineEdit" name="LineStart8">
                 <property name="styleSheet">
                  <string notr="true">padding: 1px;
border-style: solid;
border: 2px solid gray;
border-radius: 8px;</string>
                 </property>
                </widget>
               </item>
               <item row="2" column="0">
                <widget class="QLineEdit" name="LineStart7">
                 <property name="styleSheet">
                  <string notr="true">padding: 1px;
border-style: solid;
border: 2px solid gray;
border-radius: 8px;</string>
                 </property>
                </widget>
               </item>
               <item row="1" column="1">
                <widget class="QLineEdit" name="LineStart5">
                 <property name="styleSheet">
                  <string notr="true">padding: 1px;
border-style: solid;
border: 2px solid gray;
border-radius: 8px;</string>
                 </property>
                </widget>
               </item>
               <item row="1" column="0">
                <widget class="QLineEdit" name="LineStart4">
                 <property name="styleSheet">
                  <string notr="true">padding: 1px;
border-style: solid;
border: 2px solid gray;
border-radius: 8px;</string>
                 </property>
                </widget>
               </item>
               <item row="2" column="2">
                <widget class="QLineEdit" name="LineStart9">
                 <property name="styleSheet">
                  <string notr="true">padding: 1px;
border-style: solid;
border: 2px solid gray;
border-radius: 8px;</string>
                 </property>
                </widget>
               </item>
               <item row="1" column="2">
                <widget class="QLineEdit" name="LineStart6">
                 <property name="styleSheet">
                  <string notr="true">padding: 1px;
border-style: solid;
border: 2px solid gray;
border-radius: 8px;</string>
                 </property>
                </widget>
               </item>
               <item row="0" column="2">
                <widget class="QLineEdit" name="LineStart3">
                 <property name="styleSheet">
                  <string notr="true">padding: 1px;
border-style: solid;
border: 2px solid gray;
border-radius: 8px;</string>
                 </property>
                </widget>
               </item>
               <item row="0" column="1">
                <widget class="QLineEdit" name="LineStart2">
                 <property name="styleSheet">
                  <string notr="true">padding: 1px;
border-style: solid;
border: 2px solid gray;
border-radius: 8px;</string>
                 </property>
                </widget>
               </item>
               <item row="0" column="0">
                <widget class="QLineEdit" name="LineStart1">
                 <property name="styleSheet">
                  <string notr="true">padding: 1px;
border-style: solid;
border: 2px solid gray;
border-radius: 8px;</string>
                 </property>
                 <property name="text">
                  <string/>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
            </layout>
           </item>
          </layout>
          <zorder></zorder>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="groupBox_6">
          <property name="title">
           <string>The target</string>
          </property>
          <layout class="QGridLayout" name="gridLayout_16">
           <item row="0" column="0">
            <layout class="QVBoxLayout" name="verticalLayout_8">
             <item>
              <layout class="QGridLayout" name="gridLayout_17">
               <item row="2" column="1">
                <widget class="QLineEdit" name="LineTarget8">
                 <property name="styleSheet">
                  <string notr="true">padding: 1px;
border-style: solid;
border: 2px solid gray;
border-radius: 8px;</string>
                 </property>
                </widget>
               </item>
               <item row="2" column="0">
                <widget class="QLineEdit" name="LineTarget7">
                 <property name="styleSheet">
                  <string notr="true">padding: 1px;
border-style: solid;
border: 2px solid gray;
border-radius: 8px;</string>
                 </property>
                </widget>
               </item>
               <item row="2" column="2">
                <widget class="QLineEdit" name="LineTarget9">
                 <property name="styleSheet">
                  <string notr="true">padding: 1px;
border-style: solid;
border: 2px solid gray;
border-radius: 8px;</string>
                 </property>
                </widget>
               </item>
               <item row="1" column="1">
                <widget class="QLineEdit" name="LineTarget5">
                 <property name="styleSheet">
                  <string notr="true">padding: 1px;
border-style: solid;
border: 2px solid gray;
border-radius: 8px;</string>
                 </property>
                </widget>
               </item>
               <item row="1" column="0">
                <widget class="QLineEdit" name="LineTarget4">
                 <property name="styleSheet">
                  <string notr="true">padding: 1px;
border-style: solid;
border: 2px solid gray;
border-radius: 8px;</string>
                 </property>
                </widget>
               </item>
               <item row="1" column="2">
                <widget class="QLineEdit" name="LineTarget6">
                 <property name="styleSheet">
                  <string notr="true">padding: 1px;
border-style: solid;
border: 2px solid gray;
border-radius: 8px;</string>
                 </property>
                </widget>
               </item>
               <item row="0" column="2">
                <widget class="QLineEdit" name="LineTarget3">
                 <property name="styleSheet">
                  <string notr="true">padding: 1px;
border-style: solid;
border: 2px solid gray;
border-radius: 8px;</string>
                 </property>
                </widget>
               </item>
               <item row="0" column="1">
                <widget class="QLineEdit" name="LineTarget2">
                 <property name="styleSheet">
                  <string notr="true">padding: 1px;
border-style: solid;
border: 2px solid gray;
border-radius: 8px;</string>
                 </property>
                </widget>
               </item>
               <item row="0" column="0">
                <widget class="QLineEdit" name="LineTarget1">
                 <property name="styleSheet">
                  <string notr="true">padding: 1px;
border-style: solid;
border: 2px solid gray;
border-radius: 8px;</string>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
            </layout>
           </item>
          </layout>
          <zorder></zorder>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout">
          <item>
           <widget class="QPushButton" name="pushButtonSolve">
            <property name="styleSheet">
             <string notr="true">color: white;
background-color: QLinearGradient( x1: 0, y1: 0, x2: 0, y2: 1, stop: 0 #88d, stop: 0.1 #99e, stop: 0.49 #77c, stop: 0.5 #66b, stop: 1 #77c);
border-width: 1px;
border-color: #339;
border-style: solid;
border-radius: 7;
padding: 3px;
font-size: 10px;
padding-left: 5px;
padding-right: 5px;
min-width: 50px;
max-width: 50px;
min-height: 13px;
max-height: 13px;
</string>
            </property>
            <property name="text">
             <string>Solve</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="pushButtonExit">
            <property name="styleSheet">
             <string notr="true">color: white;
background-color: QLinearGradient( x1: 0, y1: 0, x2: 0, y2: 1, stop: 0 #88d, stop: 0.1 #99e, stop: 0.49 #77c, stop: 0.5 #66b, stop: 1 #77c);
border-width: 1px;
border-color: #339;
border-style: solid;
border-radius: 7;
padding: 3px;
font-size: 10px;
padding-left: 5px;
padding-right: 5px;
min-width: 50px;
max-width: 50px;
min-height: 13px;
max-height: 13px;
</string>
            </property>
            <property name="text">
             <string>Exit</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
      <widget class="QFrame" name="movesFram">
       <property name="geometry">
        <rect>
         <x>210</x>
         <y>90</y>
         <width>154</width>
         <height>142</height>
        </rect>
       </property>
       <property name="styleSheet">
        <string notr="true">background-color: rgb(255, 255, 255);
padding: 1px;
border-style: solid;
border: 2px solid gray;
border-radius: 8px;</string>
       </property>
       <property name="frameShape">
        <enum>QFrame::StyledPanel</enum>
       </property>
       <property name="frameShadow">
        <enum>QFrame::Raised</enum>
       </property>
       <layout class="QGridLayout" name="gridLayout_19">
        <item row="0" column="0">
         <layout class="QGridLayout" name="gridLayout_18">
          <item row="0" column="0">
           <widget class="QLabel" name="moveLabel1">
            <property name="styleSheet">
             <string notr="true">font-weight: bold;
color: white;
background-color: QLinearGradient( x1: 0, y1: 0, x2: 0, y2: 1, stop: 0 #88d, stop: 0.1 #99e, stop: 0.49 #77c, stop: 0.5 #66b, stop: 1 #77c);
border-width: 1px;
border-color: #339;
border-style: solid;
border-radius: 7;
padding: 3px;
font-size: 15px;
padding-left: 5px;
padding-right: 5px;
min-width: 28px;
max-width: 28px;
min-height: 28px;
max-height: 28px;
</string>
            </property>
            <property name="text">
             <string/>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QLabel" name="moveLabel2">
            <property name="styleSheet">
             <string notr="true">font-weight: bold;
color: white;
background-color: QLinearGradient( x1: 0, y1: 0, x2: 0, y2: 1, stop: 0 #88d, stop: 0.1 #99e, stop: 0.49 #77c, stop: 0.5 #66b, stop: 1 #77c);
border-width: 1px;
border-color: #339;
border-style: solid;
border-radius: 7;
padding: 3px;
font-size: 15px;
padding-left: 5px;
padding-right: 5px;
min-width: 28px;
max-width: 28px;
min-height: 28px;
max-height: 28px;
</string>
            </property>
            <property name="text">
             <string/>
            </property>
           </widget>
          </item>
          <item row="0" column="2">
           <widget class="QLabel" name="moveLabel3">
            <property name="styleSheet">
             <string notr="true">font-weight: bold;
color: white;
background-color: QLinearGradient( x1: 0, y1: 0, x2: 0, y2: 1, stop: 0 #88d, stop: 0.1 #99e, stop: 0.49 #77c, stop: 0.5 #66b, stop: 1 #77c);
border-width: 1px;
border-color: #339;
border-style: solid;
border-radius: 7;
padding: 3px;
font-size: 15px;
padding-left: 5px;
padding-right: 5px;
min-width: 28px;
max-width: 28px;
min-height: 28px;
max-height: 28px;
</string>
            </property>
            <property name="text">
             <string/>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="moveLabel4">
            <property name="styleSheet">
             <string notr="true">font-weight: bold;
color: white;
background-color: QLinearGradient( x1: 0, y1: 0, x2: 0, y2: 1, stop: 0 #88d, stop: 0.1 #99e, stop: 0.49 #77c, stop: 0.5 #66b, stop: 1 #77c);
border-width: 1px;
border-color: #339;
border-style: solid;
border-radius: 7;
padding: 3px;
font-size: 15px;
padding-left: 5px;
padding-right: 5px;
min-width: 28px;
max-width: 28px;
min-height: 28px;
max-height: 28px;
</string>
            </property>
            <property name="text">
             <string/>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QLabel" name="moveLabel5">
            <property name="styleSheet">
             <string notr="true">font-weight: bold;
color: white;
background-color: QLinearGradient( x1: 0, y1: 0, x2: 0, y2: 1, stop: 0 #88d, stop: 0.1 #99e, stop: 0.49 #77c, stop: 0.5 #66b, stop: 1 #77c);
border-width: 1px;
border-color: #339;
border-style: solid;
border-radius: 7;
padding: 3px;
font-size: 15px;
padding-left: 5px;
padding-right: 5px;
min-width: 28px;
max-width: 28px;
min-height: 28px;
max-height: 28px;
</string>
            </property>
            <property name="text">
             <string/>
            </property>
           </widget>
          </item>
          <item row="1" column="2">
           <widget class="QLabel" name="moveLabel6">
            <property name="styleSheet">
             <string notr="true">font-weight: bold;
color: white;
background-color: QLinearGradient( x1: 0, y1: 0, x2: 0, y2: 1, stop: 0 #88d, stop: 0.1 #99e, stop: 0.49 #77c, stop: 0.5 #66b, stop: 1 #77c);
border-width: 1px;
border-color: #339;
border-style: solid;
border-radius: 7;
padding: 3px;
font-size: 15px;
padding-left: 5px;
padding-right: 5px;
min-width: 28px;
max-width: 28px;
min-height: 28px;
max-height: 28px;
</string>
            </property>
            <property name="text">
             <string/>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="moveLabel7">
            <property name="styleSheet">
             <string notr="true">font-weight: bold;
color: white;
background-color: QLinearGradient( x1: 0, y1: 0, x2: 0, y2: 1, stop: 0 #88d, stop: 0.1 #99e, stop: 0.49 #77c, stop: 0.5 #66b, stop: 1 #77c);
border-width: 1px;
border-color: #339;
border-style: solid;
border-radius: 7;
padding: 3px;
font-size: 15px;
padding-left: 5px;
padding-right: 5px;
min-width: 28px;
max-width: 28px;
min-height: 28px;
max-height: 28px;
</string>
            </property>
            <property name="text">
             <string/>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QLabel" name="moveLabel8">
            <property name="styleSheet">
             <string notr="true">font-weight: bold;
color: white;
background-color: QLinearGradient( x1: 0, y1: 0, x2: 0, y2: 1, stop: 0 #88d, stop: 0.1 #99e, stop: 0.49 #77c, stop: 0.5 #66b, stop: 1 #77c);
border-width: 1px;
border-color: #339;
border-style: solid;
border-radius: 7;
padding: 3px;
font-size: 15px;
padding-left: 5px;
padding-right: 5px;
min-width: 28px;
max-width: 28px;
min-height: 28px;
max-height: 28px;
</string>
            </property>
            <property name="text">
             <string/>
            </property>
           </widget>
          </item>
          <item row="2" column="2">
           <widget class="QLabel" name="moveLabel9">
            <property name="styleSheet">
             <string notr="true">font-weight: bold;
color: white;
background-color: QLinearGradient( x1: 0, y1: 0, x2: 0, y2: 1, stop: 0 #88d, stop: 0.1 #99e, stop: 0.49 #77c, stop: 0.5 #66b, stop: 1 #77c);
border-width: 1px;
border-color: #339;
border-style: solid;
border-radius: 7;
padding: 3px;
font-size: 15px;
padding-left: 5px;
padding-right: 5px;
min-width: 28px;
max-width: 28px;
min-height: 28px;
max-height: 28px;
</string>
            </property>
            <property name="text">
             <string/>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
      <widget class="QPushButton" name="pushButtonNext">
       <property name="geometry">
        <rect>
         <x>320</x>
         <y>240</y>
         <width>37</width>
         <height>24</height>
        </rect>
       </property>
       <property name="styleSheet">
        <string notr="true">color: white;
background-color: QLinearGradient( x1: 0, y1: 0, x2: 0, y2: 1, stop: 0 #88d, stop: 0.1 #99e, stop: 0.49 #77c, stop: 0.5 #66b, stop: 1 #77c);
border-width: 1px;
border-color: #339;
border-style: solid;
border-radius: 7;
padding: 3px;
font-size: 15px;
padding-left: 5px;
padding-right: 5px;
min-width: 25px;
max-width: 25px;
min-height: 16px;
max-height: 16px;
</string>
       </property>
       <property name="text">
        <string>▶</string>
       </property>
      </widget>
      <widget class="QPushButton" name="pushButtonBack">
       <property name="geometry">
        <rect>
         <x>220</x>
         <y>240</y>
         <width>37</width>
         <height>24</height>
        </rect>
       </property>
       <property name="styleSheet">
        <string notr="true">color: white;
background-color: QLinearGradient( x1: 0, y1: 0, x2: 0, y2: 1, stop: 0 #88d, stop: 0.1 #99e, stop: 0.49 #77c, stop: 0.5 #66b, stop: 1 #77c);
border-width: 1px;
border-color: #339;
border-style: solid;
border-radius: 7;
padding: 3px;
font-size: 15px;
padding-left: 5px;
padding-right: 5px;
min-width: 25px;
max-width: 25px;
min-height: 16px;
max-height: 16px;
</string>
       </property>
       <property name="text">
        <string>◀</string>
       </property>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>404</width>
     <height>19</height>
    </rect>
   </property>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
